src/Microsoft.OData.Core/ODataCollectionWriterCore.cs

```diff
@@ -224,10 +226,10 @@ public sealed override void WriteEnd()
         /// Asynchronously finish writing a collection.
         /// </summary>
         /// <returns>A task instance that represents the asynchronous write operation.</returns>
-        public sealed override Task WriteEndAsync()
+        public sealed override async Task WriteEndAsync()
         {
             this.VerifyCanWriteEnd(false);
-            return TaskUtils.GetTaskForSynchronousOperation(this.WriteEndImplementation)
+            await this.WriteEndImplementationAsync()

```

Comment:
Elide

Reply:
Explained why it's not particularly recommendable to do that here https://github.com/OData/odata.net/pull/2059#discussion_r635801748